internal/core/notebook.go

```diff
@@ -148,6 +150,15 @@ func (n *Notebook) NewNote(opts NewNoteOpts) (*Note, error) {
 		return nil, wrap(err)
 	}
 
+	var idGenerator IDGenerator
+	if opts.ID != "" {
```

Comment:
The command parsing was confusing me. I was assuming that null values are converted to their type as an empty value. I might be way off though and need to change this.

Reply:
That's right, there's no `null` values except for pointer types. As you (rightly) used `ID string` instead of `ID *string`, checking for `""` is the correct approach.

You could have used my `opt.String` type as an alternative, but to be honest this is not very idiomatic in Go. I added it as frustration that a real optional Go type was missing. This might change in the new Go 1.18 which added generics.